src/entities/text-parsing/lib/helpers/find-parent-key-by-id.ts

```diff
@@ -3,7 +3,10 @@ import type {
   GetAttendanceOptionsAll,
 } from '@/entities/attendance-option/@x/text-parsing'
 
-export const findParentKeyById = (id: number, attendanceOptions: GetAttendanceOptionsAll) => {
+export const findParentKeyById = (
+  id: number,
+  attendanceOptions: GetAttendanceOptionsAll,
+): AttendanceOptionKey | null => {
   let result = null
 
   Object.keys(attendanceOptions).forEach((key) => {
```

Comment:
[제안]
현재 let 으로 선언한 result 변수가 any 타입으로 추론되고 있습니다. 반환 타입을 명시해주셔서 크게 문제되지는 않지만 명확한 타입을 정해주시거나 아래와 같은 로직으로 수정하는 것이 추후 유지보수 시점에 유리할 것 같은데 어떠신가요?

<br />

```suggestion
  return Object.keys(attendanceOptions).reduce<AttendanceOptionKey | null>((result, key) => {
    if (
      attendanceOptions[key as AttendanceOptionKey].detailOptions.some((option) => option.id === id)
    ) {
      return key as AttendanceOptionKey
    }

    return result
  }, null)
```

Reply:
제안 감사합니다.🙂 혹시 reduce 함수 대신 `for ... of` 구문을 사용하는 것은 어떠실까요? 이 함수는 불변성이 크게 중요한 로직이 아니기도 하고, 조기 종료가 가능하다는 점과 가독성이 조금 더 좋게 느껴져서요. 

```tsx
  for (const key of Object.keys(attendanceOptions) as AttendanceOptionKey[]) {
    if (attendanceOptions[key].detailOptions.some((option) => option.id === id)) {
      return key
    }
  }

  return null
```